src/v/redpanda/application.cc

```diff
@@ -241,6 +241,16 @@ void application::hydrate_config(const po::variables_map& cfg) {
             vassert(!kafka_api.empty(), "There are no kafka_api listeners");
             _proxy_client_config->brokers.set_value(
               std::vector<unresolved_address>{kafka_api[0].address});
+            const auto& kafka_api_tls
+              = config::shard_local_cfg().kafka_api_tls.value();
+            if (!kafka_api_tls.empty()) {
+                vassert(
+                  kafka_api[0].name == kafka_api_tls[0].name,
+                  "pandaproxy_client.broker_tls could not be inferred. Please "
+                  "configure manually.");
```

Comment:
should we also print the configuration of the kafka_api_tls? `"...could not be inferred from the kafka_api_tls: {}", kafka_api_tls`